triton/stream_reader.go

```diff
@@ -37,13 +37,19 @@ func (msr *multiShardStreamReader) Checkpoint() (err error) {
 	return
```

Comment:
Oh because the `ShardStreamReader` struct also has a LastSequenceNumber.  Yeah, I think they are a little redundant.  

I think it would be good to change the triton service interface to only expose a client side reader and the aggregation service and then make everything else private.  I didn't want to remove it because I wasn't sure what part of the public API is exported.